File: after/ftplugin/javascript/decodeURIComponent.js
"use strict";
process.stdin.setEncoding("utf8");

process.stdin.once("readable", () => {
  const chunk = process.stdin.read();
  if (chunk !== null) {
    let result = decodeURIComponent(chunk);
    if (result) {
      process.stdout.write(result);
    }
  }
});

File: after/ftplugin/javascript/decodeURIComponent.py
#!/usr/bin/env python3

import sys
import urllib.parse

# Translates %xx escapes to single letter equivalent,
# e.g. "search+query%20%28correct%29" to "search+query (correct)"
# Note: Intended to be used as a filter in Vim, so it reads stdin and writes to stdout

sys.stdout.write(urllib.parse.unquote(sys.stdin.read()))
